### PR TITLE
Various doc fixes

### PR DESCRIPTION
**Pull request by goderbauer**
## Description

Takes the doc fixes from https://github.com/flutter/flutter/pull/33460 and applies them without making the analyzer mad. Also filed https://github.com/flutter/flutter/issues/35547 to allow "Sample Code" headings in the future and move the discussion around those to the issue.

/cc @Hixie @gspencergoog 

## Related Issues

None

## Tests

I added the following tests:

None, doc only change.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/container.dart [hunk 1]
```diff
@@ -44,6 +44,50 @@ import 'image.dart';
...
 /// {@end-tool}
 ///
+/// ## Layout behavior
+///
+/// _See [BoxConstraints] for an introduction to box layout models._
+///
+/// Since [Container] combines a number of other widgets each with their own
+/// layout behavior, [Container]'s layout behavior is somewhat complicated.
+///
+/// tl;dr: [Container] tries, in order: to honor [alignment], to size itself to
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/container.dart]: Is tl;dr ok in API docs? Ref: #18542

File: packages/flutter/lib/src/widgets/container.dart [hunk 2]
```diff
@@ -44,6 +44,50 @@ import 'image.dart';
...
+/// the [child], to honor the `width`, `height`, and [constraints], to expand to
+/// fit the parent, to be as small as possible.
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/container.dart]:
```suggestion
/// fit the parent, and to be as small as possible.
```

File: packages/flutter/lib/src/widgets/container.dart [hunk 3]
```diff
@@ -44,6 +44,50 @@ import 'image.dart';
...
+/// [constraints], and no [alignment], and the [Container] passes the
+/// constraints from the parent to the child and sizes itself to match the
+/// child.
+///
+/// The [margin] and [padding] properties also affect the layout, as described
+/// in the documentation for those properties. (Their effects merely augment the
+/// rules described above.) The [decoration] can implicitly increase the
+/// [padding] (e.g. borders in a [BoxDecoration] contribute to the [padding]);
+/// see [Decoration.padding].
+///
```
Comment (Piinks) [inline packages/flutter/lib/src/widgets/container.dart]: Would a diagram for any of these scenarios listed above be beneficial in illustrating the point? Not sure.

**Review by Piinks — APPROVED** [review]
LGTM +/- nits below. :)

File: packages/flutter/lib/src/widgets/container.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/container.dart]: Fixed.

File: packages/flutter/lib/src/widgets/container.dart (hunk 2 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/container.dart]: The and seems to make it ambitious.

File: packages/flutter/lib/src/widgets/container.dart (hunk 3 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/container.dart]: I'll leave that for a follow-up :)

**Comment by goderbauer** [comment]
On cirrus it shows that all checks passed: https://cirrus-ci.com/build/5674139195539456

Not sure why Github is disagreeing. Submitting.

**Pull request merged**